mypy/test/testipc.py

```diff
@@ -0,0 +1,36 @@
+import os
+import sys
+import time
+from unittest import TestCase, main
+from multiprocessing import Process
+
+from mypy.ipc import IPCClient, IPCServer, IPCException
+
+
+if sys.platform == 'win32':
+    CONNECTION_NAME = r'\\.\pipe\dmypy-ipc-test-{}.pipe'
+else:
+    CONNECTION_NAME = 'dmypy-test-ipc-{}.sock'
+
+
+def server(msg: str, pid: int) -> None:
+    with IPCServer(CONNECTION_NAME.format(pid)) as server:
+        server.read()
+        server.write(msg.encode())
+        server.cleanup()
+
+
+class IPCTests(TestCase):
+    def test_transaction_large(self) -> None:
+        pid = os.getpid()
+        msg = 't' * 100001  # longer than the max read size of 100_000
+        p = Process(target=server, args=(msg, pid), daemon=True)
+        p.start()
+        time.sleep(1)  # wait for server to get set up
```

Comment:
Solve, don't evade, concurrency problems.

Probably the thing to do is have a `Semaphore` or `Event` that gets signaled once the server is ready.

Reply:
Oh yes, this was just a test to see if that was the root cause, and it was not. What I really should do is pass the connection name from the server process to the client/test process to set up the client correctly.